tools/chapel-py/src/method-tables/core-methods.h

```diff
@@ -143,7 +143,10 @@ CLASS_BEGIN(Error)
   PLAIN_GETTER(Error, kind, "Retrieve the kind ('error', 'warning') of this type of error",
                const char*, return chpl::ErrorBase::getKindName(node->kind()))
   PLAIN_GETTER(Error, type, "Retrieve the unique name of this type of error",
-               const char*, return chpl::ErrorBase::getTypeName(node->type()))
+               std::optional<const char*>,
+               const char* name = chpl::ErrorBase::getTypeName(node->type());
+               return name ? std::optional(name) : std::nullopt;
```

Comment:
I think you can just return `{ name }` or even `name`; it will get auto-wrapped.

Reply:
Do you mean `return name ? name : std:nullopt` or `return name`? The former fails to compile and I am not a fan of the latter

Reply:
the former; `return name ? { name } : std::nullopt` should work (I hope you mean "failed to compile" not because of the `std:nullopt` typo 😃 )

Reply:
No it fails to compile: "incompatible operand types ('const char *' and 'const nullopt_t')"

Note that the following does work
```
if (name) return name; else return std::nullopt;
```

But I think the ternary operator is better in this case